src/tekton/startpipelinefromyaml.ts

```diff
@@ -0,0 +1,139 @@
+    pipeline = JSON.parse(result.stdout);
+  } catch (ignore) {
+    //show no pipelines if output is not correct json
+  }
+  return getPipelineRunData(pipeline, pipelineRunData);
+}
+
+function getPipelineRunData(pipeline: TknPipelineTrigger, pipelineRunData: PipelineRunKind): PipelineRunKind {
+  const pipelineName = pipeline.metadata.name;
+  const resources = pipelineRunData?.spec.resources;
+  const workspaces = pipelineRunData?.spec.workspaces;
+  const params = pipelineRunData?.spec.params;
+  const serviceAccountName = pipelineRunData?.spec.serviceAccountName;
+
+  const newPipelineRun = {
+    apiVersion: pipeline.apiVersion,
+    kind: PipelineRunModel.kind,
+    metadata: {
+      generateName: `${pipelineName}-`,
+      namespace: pipeline.metadata.namespace,
+    },
+    spec: {
+      pipelineRef: {
+        name: pipelineName,
+      },
+      resources,
+      params,
+      workspaces,
+      status: null,
+      serviceAccountName
+    },
+  };
+  return newPipelineRun;
+}
+
+function getPipelineRunResources(resources: Resources[]): Resources[] {
+  const newResource = [];
+  if (resources && resources.length !== 0) {
+    resources.map(val => {
+      newResource.push({name: val.name, resourceRef: { name: val.resourceRef}})
+    })
+  }
+  return newResource;
+}
+
+function getPipelineRunParams(params: Params[]): Params[] {
+  if (params && params.length !== 0) {
+    params.map(val => {
+      val['value'] = val.default;
+      delete val.default
+    });
+    return params;
+  } else {
+    return [];
+  }
+}
+
+function getPipelineRunWorkspaces(workspace: Workspaces[], volumeClaimTemplate: VCT[]): PipelineRunWorkspace[] {
+  const newWorkSpace = [];
+  if (workspace && workspace.length !== 0) {
+    workspace.map(value => {
+      const workspaceObject = {};
+      workspaceObject['name'] = value.name;
+      if (value.workspaceType === 'Secret') {
+        workspaceObject['secret'] = {
+          items: value.item,
+          secretName: value.workspaceName
+        }
+      } else if (value.workspaceType === 'ConfigMap') { 
+        workspaceObject['configMap'] = {
+          items: value.item,
+          name: value.workspaceName
+        }
+      } else if (value.workspaceType === 'PersistentVolumeClaim') { 
+        workspaceObject['persistentVolumeClaim'] = {
+          claimName: value.workspaceName
+        }
+      } else if (value.workspaceType === 'EmptyDirectory') { 
+        workspaceObject['persistentVolumeClaim'] = {
+          claimName: value.workspaceName,
+          emptyDir: {}
+        }
+      }
+      newWorkSpace.push(workspaceObject);
+    })
+  }
+  if (volumeClaimTemplate && volumeClaimTemplate.length !== 0) {
+    volumeClaimTemplate.map(value => {
+      const workspaceObject = {};
+      workspaceObject['name'] = value.metadata.name,
+      workspaceObject[value.kind] = {
+        spec: value.spec
+      }
+      newWorkSpace.push(workspaceObject);
+    })
+  }
+  return newWorkSpace;
+}
```

Comment:
It is better to rename this file to `start-pipeline-from-json.ts`, as it seems that it works with JSON not YAML.